bika/lims/browser/fields/aranalysesfield.py

```diff
@@ -243,16 +251,23 @@ def _to_service(self, thing):
         logger.warn(msg)
         return None
 
-    def _is_frozen(self, brain_or_object):
-        """Check if the passed in object is frozen
-
-        :param obj: Analysis or AR Brain/Object
+    def _is_frozen(self, brain_or_object, additional_frozen_actions=None):
+        """Check if the passed in object is frozen: the objectt is cancelled,
+        inactive or has been verified at some point
+        :param brain_or_object: Analysis or AR Brain/Object
+        :param additional_frozen_actions: additional actions to check against
         :returns: True if the object is frozen
         """
-        obj = api.get_object(brain_or_object)
-        active = api.is_active(obj)
-        verified = wasTransitionPerformed(obj, 'verify')
-        return not active or verified
+        if not api.is_active(brain_or_object):
+            return True
+        actions = getReviewHistoryActionsList(api.get_object(brain_or_object))
+        if 'verify' in actions:
+            return True
+        additional = additional_frozen_actions or list()
+        for action in additional:
+            if action in actions:
+                return True
+        return False
```

Comment:
Hi Jordi, I was just curious to check other approaches. Feel free to adopt

```python
def _is_frozen(self, brain_or_object, *frozen_transitions):
    """Check if the passed in object is frozen: the object is cancelled,
        inactive or has been verified at some point
        :param brain_or_object: Analysis or AR Brain/Object
        :param frozen_actions: additional transitions that freeze the object
        :returns: True if the object is frozen
    """
    if api.is_active(brain_or_object):
        return False
    frozen_transitions = set("verified")
    # append additional frozen transitions
    frozen_transitions.add(*frozen_transitions)
    performed_transitions = set(getReviewHistoryActionsList(api.get_object(brain_or_object)))
    if not frozen.intersection(performed_transitions):
        return False
    return True
```

Now you can call it like this:
```python
if self._is_frozen(analysis, "retract", "verified", ...):
    logger.warn("Inactive/verified/retracted Analyses can not be removed.")
    continue
```